tests/integration/networkInstance_test.go

```diff
@@ -0,0 +1,214 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/lf-edge/eden/pkg/cloud"
+	"github.com/lf-edge/eden/pkg/device"
+	"github.com/lf-edge/eden/pkg/einfo"
+	"github.com/lf-edge/eve/api/go/config"
+	"testing"
+)
+
+func TestNetworkInstanceSwitch(t *testing.T) {
+	ctx, devUUID, err := adamPrepare()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	niID := "eab8761b-5f89-4e0b-b757-4b87a9fa93e1"
+
+	cloudCxt := &cloud.Ctx{}
+	err = cloudCxt.AddNetworkInstanceConfig(&config.NetworkInstanceConfig{
+		Uuidandversion: &config.UUIDandVersion{
+			Uuid:    niID,
+			Version: "4",
+		},
+		Displayname: "testSwitch",
+		InstType:    config.ZNetworkInstType_ZnetInstSwitch,
+		Activate:    true,
+		Port: &config.Adapter{
+			Type: config.PhyIoType_PhyIoNoop,
+			Name: "uplink",
+		},
+		Cfg:    &config.NetworkInstanceOpaqueConfig{},
+		IpType: config.AddressType_IPV4,
+		Ip: &config.Ipspec{
+			Dhcp:    config.DHCPType_DHCPNoop,
+			Subnet:  "10.1.0.0/16",
+			Gateway: "10.1.0.1",
+			Domain:  "",
+			Ntp:     "",
+			Dns:     []string{"10.1.0.1"},
+			DhcpRange: &config.IpRange{
+				Start: "10.1.0.2",
+				End:   "10.1.255.254",
+			},
+		},
+		Dns: nil,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	deviceCtx := device.CreateWithBaseConfig(*devUUID, cloudCxt)
+	deviceCtx.SetNetworkInstanceConfig([]string{niID})
+	b, err := deviceCtx.GenerateJSONBytes()
+	if err != nil {
+		t.Fatal(err)
+	}
+	configToSet := fmt.Sprintf("%s", string(b))
+	t.Log(configToSet)
+	res, err := ctx.ConfigSet(devUUID.String(), configToSet)
+	if err != nil {
+		t.Log(res)
+		t.Fatal(err)
+	}
```

Comment:
Same comments as for the first tests apply here as well.

Reply:
TableDriven variant: https://github.com/lf-edge/eden/blob/90d45b9511149229f28d60d2d7dd6ffb45dfe068/tests/integration/networkInstance_test.go#L108-L138
More information in log: https://github.com/lf-edge/eden/blob/90d45b9511149229f28d60d2d7dd6ffb45dfe068/tests/integration/networkInstance_test.go#L135-L157